Kommunicate/Classes/KMPushNotificationHelper.swift

```diff
@@ -0,0 +1,252 @@
+    ///   - notification: notification that is tapped.
+    public func openConversationFromListVC(_ viewController: KMConversationListViewController, notification: NotificationData) {
+        viewController.launchChat(contactId: notification.userId, groupId: notification.groupId, conversationId: notification.conversationId)
+    }
+
+    /// Returns an instance of list view controller which should be pushed from outside.
+    /// It will launch `ALKConversationViewController`.
+    ///
+    /// - NOTE: Use this to launch chat when some other screen is opened.
+    /// - Parameters:
+    ///   - notification: notification that is tapped.
+    ///   - configuration: `ALKConfiguration` object.
+    /// - Returns: An instance of `ALKConversationListViewController`
+    public func getConversationVCToLaunch(notification: NotificationData, configuration: ALKConfiguration) -> KMConversationListViewController {
+        let viewController = KMConversationListViewController(configuration: configuration)
+        viewController.contactId = notification.userId
+        viewController.conversationId = notification.conversationId
+        viewController.channelKey = notification.groupId
+        return viewController
+    }
+
+    /// Refrehses `ALKConversationViewController` for the arrived notification.
+    ///
+    /// - NOTE: Use this when `ALKConversationViewController` is at top
+    /// - Parameters:
+    ///   - viewController: An instance of `ALKConversationViewController` which is at top.
+    ///   - notification: notification that is tapped.
+    public func refreshConversation(_ viewController: KMConversationViewController, with notification: NotificationData) {
+        viewController.unsubscribingChannel()
+        viewController.viewModel.contactId = notification.userId
+        viewController.viewModel.channelKey = notification.groupId
+        var convProxy: ALConversationProxy?
+        if let convId = notification.conversationId, let conversationProxy = ALConversationService().getConversationByKey(convId) {
+            convProxy = conversationProxy
+        }
+        viewController.viewModel.conversationProxy = convProxy
+        viewController.viewWillLoadFromTappingOnNotification()
+        viewController.refreshViewController()
+    }
+
+    /// Checks whether Applozic ViewController is at top.
+    ///
+    /// - WARNING: Doesn't work if Applozic's Controller is added inside some container.
+    /// - Returns: Bool value indicating whether Applozic view is at top.
+    public func isApplozicVCAtTop() -> Bool {
+        guard let topVC = ALPushAssist().topViewController else { return false }
+        let topVCName = String(describing: topVC.classForCoder)
+        switch topVCName {
+        case "MuteConversationViewController",
+             "ALKWebViewController",
+             "SelectProfilePicViewController",
+             "CAMImagePickerCameraViewController",
```

Comment:
We don't have to check for internal classes, we can use ApplozicSwift's helper for that.